### ISSUE TITLE
Permissions: datasources:query missing for user with no basic permissions

### ISSUE THREAD

**Issue by slhck** [grafana/grafana#80129]
### What happened?

I want to set up an organization with one team, which has its own dashboard in a separate folder. This team should not have access to the dashboards in the other folders.

To accomplish that, I add users with *No basic permission* and add them to the organization. Then, I add them as *Member* to the respective new team.

The team then gets *Viewer* permissions on the dashboard folder that contains the team's dashboards.

A member of the team however cannot view the dashboard, with a missing "datasources:query" permission and "annotations:read" permission:

> Error
> You'll need additional permissions to perform this action. Permissions needed: datasources:query

> Error
> You'll need additional permissions to perform this action. Permissions needed: annotations:read

### What did you expect to happen?

I expected the team members to gain the respective permissions needed to view the dashboards in the folder from which they inherited the *Viewer* permission.

### Did this work before?

I have not tried this before.

### How do we reproduce it?

1. Create a new organization
2. Create a dashboard in a subfolder (e.g. "test")
3. Create a team called "test"
4. Add a new user with *No basic permissions* assigned to the organization
5. Add the user to the team with *Member* permissions
6. Add the team to the "test" subfolder with *Viewer* permissions
7. Log in as the user
8. The user can see the folder and the dashboard, but cannot view its contents — with error popups

### Environment (with versions)?

Grafana: Grafana v10.2.3 (1e84fede54)
OS: macOS 12
Browser: Chrome (latest)

### Grafana platform?

Docker

**Comment by slhck** [comment]
PS: I can accomplish this in a more convoluted way by removing the general *View* permission from the *Viewer* role from all folders, and explcitly creating a general team that contains all users of the organization (minus the special team referred to in the issue above). I give this one Viewer permission on all dashboard folders.

**Comment by AutonomicMattRoy** [comment]
@slhck Were you able to find any solutions to this? Running into the same issue, seemingly since the `None` role is stripped of all permissions and the folder/dashboard permissions for a given team don't add `datasources:query` and `annotations:read` which is needed to actually get any data on a dashboard that can be viewed.

**Comment by slhck** [comment]
No, that is precisely the issue, and the only workaround I have is the one I described above, namely to explicitly add all users to a team and give that the viewer permission.

**Comment by marefr** [comment]
@grafana/identity-squad please triage

**Comment by shurkus** [comment]
have same issue with renderer when create image
![test](https://github.com/grafana/grafana/assets/10572093/4942f652-1891-4293-afd8-8235b0943364)

**Comment by omar-sadiqi** [comment]
Hello @marefr @slhck 
we are actually using grafana docker Grafana v10.4.0 (03f502a94d)
after creating an basic user with no permission and given him the view permission over a dashboard , we can see the dashboard by multiples alerts are showing in the ui including 

 Templating [SchemaName]
Error updating options: error when executing the sql query

Templating [Status]
Error updating options: error when executing the sql query

Templating [DatasetName]
Error updating options: error when executing the sql query

Templating [LoadingType]
Error updating options: error when executing the sql query

Error
You'll need additional permissions to perform this action. Permissions needed: annotations:read

Error
You'll need additional permissions to perform this action. Permissions needed: alert.rules:read


Any way to fix it , as i'm ready to work on it

**Comment by gcapozzo** [comment]
Same issue here, continues in 11.1.4, workaround adding users as a viewers and removing default permission of view in every folder/dashboard.

**Comment by joerg1985** [comment]
Is the limitation of users to specific dashboards a strong encapsulation?
I have given permission to the dashboards by teams, so a user can only see some dashboards.
When adding users to the viewer role, will they be able to query the data of the other dashboards when calling the backend directly?

**Comment by inih** [comment]
Same issue here, currently on version 11.3.1.

**Comment by camerkel** [comment]
It doesn't make sense to be able to assign a user `No basic permission`, if you are not able to re-grant the wanted permissions for a certain dashboard, like `datasources:query`:

![Image](https://github.com/user-attachments/assets/246dcf5e-5b8f-4e3a-9a9a-98dd20eac2b7)

**Comment by nicolai-hornung-bl** [comment]
@camerkel its possible with custom roles (those I believe are a pure Enterprise / Grafana Cloud Feature: https://grafana.com/docs/grafana/latest/administration/roles-and-permissions/access-control/). Using them you can apply scopes to both datasources and Dashboards/Folders and give people limited access to these resources. No Basic Role is required in this case to ensure only custom roles can provide access to resources.

**Comment by joy-ibss** [comment]
> Same issue here, continues in 11.1.4, workaround adding users as a viewers and removing default permission of view in every folder/dashboard.

This one is the only solution so far.
Data 7.5.25
Grafana 11.4
OS Rocky9
I tested and it solved the issues.
Hectic work and weird issue it is.

**Comment by Jguer** [comment]
`No basic role` allows the intended functionality with datasource permissions by adding datasource query permissions to the team [docs](https://grafana.com/docs/grafana/latest/administration/data-source-management/#data-source-permissions). It's currently available in Grafana Enterprise and Cloud Free+.

![Image](https://github.com/user-attachments/assets/62d2b964-9c45-4153-9319-c5ec3187bf15)

Annotation and Alert rule permissions can also be granted to teams directly via fixed roles or custom roles, thus granting users with basic role `None` all of the required permissions.

Currently the workaround suggested by @gcapozzo is the closest to this flow achievable in self-hosted (given the users are allowed to access all datasources in the organization)

**Comment by slhck** [comment]
I don't think this is an acceptable solution If this works as intended and is not addressed any other way, it still remains incredibly confusing, as can be seen by the reactions to this post.

**Comment by Jguer** [comment]
You are right @slhck , we're looking at disabling some Role None assignment routes if the instance doesn't have access to the features to allow a proper management flow.
I'll also see if we can make it more prominent in documentation
